test/e2e/tests/request-queuing/batch-txs-per-dapp-extra-tx.spec.js

```diff
@@ -101,55 +94,59 @@ describe('Request Queuing for Multiple Dapps and Txs on different networks', fun
 
         // Dapp 1 send 2 tx
         await driver.switchToWindowWithUrl(DAPP_URL);
-        await driver.delay(largeDelayMs);
+        await driver.findElement({
+          css: '[id="chainId"]',
+          text: '0x53a',
+        });
         await driver.clickElement('#sendButton');
         await driver.clickElement('#sendButton');
 
-        await driver.delay(largeDelayMs);
+        await driver.waitUntilXWindowHandles(4);
 
         // Dapp 2 send 2 tx
         await driver.switchToWindowWithUrl(DAPP_ONE_URL);
-        await driver.delay(largeDelayMs);
+        await driver.findElement({
+          css: '[id="chainId"]',
+          text: '0x53a',
+        });
         await driver.clickElement('#sendButton');
         await driver.clickElement('#sendButton');
 
-        await driver.delay(largeDelayMs);
-
         // Dapp 1 send 1 tx
         await driver.switchToWindowWithUrl(DAPP_URL);
-        await driver.delay(largeDelayMs);
+        await driver.findElement({
+          css: '[id="chainId"]',
+          text: '0x53a',
+        });
         await driver.clickElement('#sendButton');
 
-        await switchToNotificationWindow(driver, 4);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
 
-        let navigationElement = await driver.findElement(
-          '.confirm-page-container-navigation',
+        await driver.waitForSelector(
+          By.xpath("//div[normalize-space(.)='1 of 2']"),
         );
 
-        let navigationText = await navigationElement.getText();
-
-        assert.equal(navigationText.includes('1 of 2'), true);
-
         // Reject All Transactions
         await driver.clickElement('.page-container__footer-secondary a');
 
-        await driver.clickElement({ text: 'Reject all', tag: 'button' }); // TODO: Do we want to confirm here?
+        // TODO: Do we want to confirm here?
+        await driver.clickElementAndWaitForWindowToClose({
+          text: 'Reject all',
+          tag: 'button',
+        });
 
-        // Wait for confirmations to close and transactions from the second dapp to open
-        // Large delays to wait for confirmation spam opening/closing bug.
-        await driver.delay(largeDelayMs);
+        await driver.switchToWindowWithUrl(DAPP_URL);
 
         // Wait for new confirmations queued from second dapp to open
-        await switchToNotificationWindow(driver, 4);
+        // We need a big delay to make sure dialog is not invalidated
+        // TODO: find a better way to handle different dialog ids
+        await driver.delay(2000);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
 
-        navigationElement = await driver.findElement(
-          '.confirm-page-container-navigation',
+        await driver.waitForSelector(
+          By.xpath("//div[normalize-space(.)='1 of 2']"),
         );
 
-        navigationText = await navigationElement.getText();
-
-        assert.equal(navigationText.includes('1 of 2'), true);
```

Comment:
this is a bad pattern, which adds race conditions when the element is present but inner value is not the one we expect